internal/stacks/stackruntime/internal/stackeval/main.go

```diff
@@ -597,3 +597,17 @@ func (m *Main) availableProvisioners() map[string]provisioners.Factory {
 		},
 	}
 }
+
+// PlanTimestamp provides the timestamp at which the plan
+// associated with this operation is being executed.
+// If we are planning we either take the forced timestamp or the saved current time
+// If we are applying we take the timestamp time from the plan
+func (m *Main) planTimestamp() *time.Time {
+	// TODO: Implement for applying
+	// TODO: Save the current time in m.planning
+	if m.planning != nil && m.planning.forcePlanTimestamp != nil {
+		return m.planning.forcePlanTimestamp
```

Comment:
`forcePlanTimestamp` is defined as a pointer so that it's possible for it to be `nil` in the normal case where we aren't forcing the timestamp for testing purposes, but I don't think there's any case where there is no timestamp _at all_ because we can always fall back on the current timestamp as you did below.

Therefore I think it would be sufficient for the `PlanTimestamp` methods to return a non-pointer `time.Time` value, to clarify that it's not optional in this context.